sdk/python/tests/test_util.py

```diff
@@ -14,89 +14,23 @@
-
-def get_params_from_config(pipeline_name, config_path):
-    pipelines = get_config(config_path)
-
-    for pipeline in pipelines:
-        if pipeline_name == pipeline["pipeline"]:
-            return pipeline
-
-def test_workflow_without_decorator(pipeline_mod, params_dict):
-    """Test compiling a workflow and appending pipeline params."""
-
-    try:
-        pipeline_params = []
-        for param in params_dict.get('paramsList', []):
-            pipeline_params.append(getattr(pipeline_mod, param))
-
-        compiled_workflow = compiler.TektonCompiler()._create_workflow(
-            getattr(pipeline_mod,params_dict['function']),
-            params_dict.get('name', None),
-            params_dict.get('description', None),
-            pipeline_params if pipeline_params else None,
-            params_dict.get('conf', None))
-        return True
-    except :
-        return False
-
-def test_nested_workflow(pipeline_mod, pipeline_list):
-    """Test compiling a simple workflow, and a bigger one composed from the simple one."""
-
-    tmpdir = tempfile.mkdtemp()
-    try:
-        for pipeline in pipeline_list:
-            pipeline_name = pipeline['name']
-            package_path = os.path.join(tmpdir, pipeline_name + '.zip')
-            compiler.TektonCompiler().compile(getattr(pipeline_mod, pipeline_name), package_path)
-        return True
-    except:
-        return False
-
+import os
+import unittest
+from shutil import copyfile
+from compiler.compiler_tests import TestTektonCompiler
 
 if __name__ == '__main__':
-    test_data_path = sys.argv[1]
-    config_path = sys.argv[2]
-    did_compile = False
-
-    # Import pipeline
-    test_data_dir, test_data_file = os.path.split(test_data_path)
-    import_name, test_data_ext = os.path.splitext(test_data_file)
-    sys.path.append(test_data_dir)
-    pipeline_mod = importlib.import_module(import_name)
-
-    # Get the pipeline specific parameters from the config file
-    params = get_params_from_config(test_data_file, config_path)
-    if params == None:
-        raise ValueError('No pipeline matches available in the config file')
-    test_type = params['type']
-
-    if test_type == 'nested':
-        did_compile = test_nested_workflow(pipeline_mod, params['components'])
-    elif test_type == 'no_decorator':
-        did_compile = test_workflow_without_decorator(pipeline_mod, params['components'])
+    test_name = os.path.splitext(os.path.split(sys.argv[1])[1])[0]
+    save_path = sys.argv[2]
+    golden_path = os.path.join(os.path.dirname(__file__), 'compiler/testdata', test_name+'.yaml')
+
+    if test_name == "compose":
+        test = TestTektonCompiler().test_compose
+    elif test_name == "basic_no_decorator":
+        test = TestTektonCompiler().test_basic_no_decorator
     else:
-        raise ValueError('Pipeline type \''+test_type+'\' is not recognized')
+        raise ValueError('Pipeline named \''+test_name+'\' is not recognized')
```

Comment:
instead of string concatenation with escaped single quotes, use format string or `%` operator